addons/mass_mailing/wizard/mailing_mailing_test.py

```diff
@@ -23,9 +23,72 @@ def _default_email_to(self):
             ('create_uid', '=', self.env.uid),
         ], order='create_date desc', limit=1).email_to or self.env.user.email_formatted
 
+    def _get_model_selection(self):
+        result = self.env["ir.model"].sudo().search([("transient", "=", False)])
+        return [(model.model, model.name) for model in result]
+
     email_to = fields.Text(string='Recipients', required=True,
                            help='Carriage-return-separated list of email addresses.', default=_default_email_to)
     mass_mailing_id = fields.Many2one('mailing.mailing', string='Mailing', required=True, ondelete='cascade')
+    email_from = fields.Char(string='From', related='mass_mailing_id.email_from')
```

Comment:
Email from can actually be dynamic, as it is given to the composer who is going to try to render it. So it should be rendered also, as far as I know.